oauthproxy.go

```diff
@@ -154,9 +154,28 @@ func NewReverseProxy(target *url.URL, opts *options.Options) (proxy *httputil.Re
 			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
 		}
 	}
+	setProxyErrorHandler(proxy, opts, target)
 	return proxy
 }
 
+func setProxyErrorHandler(proxy *httputil.ReverseProxy, opts *options.Options, target *url.URL) {
+	templates := loadTemplates(opts.CustomTemplatesDir)
+	proxy.ErrorHandler = func(w http.ResponseWriter, r *http.Request, proxyErr error) {
+		logger.Printf("proxying error :%v", proxyErr)
+		w.WriteHeader(http.StatusBadGateway)
+		data := struct {
+			Title       string
+			Message     string
+			ProxyPrefix string
+		}{
+			Title:       "Bad Gateway",
+			Message:     fmt.Sprintf("could not get response from upstream '%s': %v", target.String(), proxyErr),
```

Comment:
I'm unsure if we want to print the target, that could reveal internal details about the systems, what do you think?

Reply:
Good point!
But we could still leak internal network details by printing out the proxy error (such as the ip address and port of the upstream in the case of a TCP error)
What do you think?

Reply:
Don't suppose you happen to have an example of the printed logs from your manual testing? Seeing the real log would help make the decision I think.

If not I'll spin up an env later and have a try

Reply:
Here, the upstream is unreachable:

![image](https://user-images.githubusercontent.com/17098052/82759314-e3540b00-9dec-11ea-97c5-aaf7311a20da.png)

Reply:
I'd be tempted to not display the errors to the end user, instead maybe just display

`Error proxying to upstream server`

The user should be expected to report the problems to the administrator right? And they would have access to the logs in which they should find the logging of the errors right?

Reply:
I completely agree with you
Thanks for the feedback! :+1: